### ISSUE TITLE
SRP violation 

### ISSUE THREAD

**Issue by adrianpawlik** [Bubuhubu87/bc-jokes#3]
https://github.com/Bubuhubu87/bc-jokes/blob/28f5380d823112e6014773dbe4c4d93695f31daa/jokes/src/app/modules/jokes/components/categories/categories.component.ts#L51

**Comment by Bubuhubu87** [comment]
That's true, I thought about cache in joke.service.ts but I had problem with cache observable, could you give mi advice how to deal with that?

**Comment by Bubuhubu87** [comment]
To be done with AOP approach.
